APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Collectors/APCDisplacementTrackingCollector.m

```diff
@@ -0,0 +1,175 @@
+// the copyright holders even if such marks are included in this software. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+// 
+
+#import "APCAppCore.h"
+#import "APCDisplacementTrackingCollector.h"
+
+@interface APCDisplacementTrackingCollector () <CLLocationManagerDelegate>
+
+@property (strong, nonatomic) CLLocationManager *locationManager;
+@property (nonatomic, strong) CLLocation  *baseTrackingLocation;
+@property (nonatomic, strong) CLLocation  *mostRecentUpdatedLocation;
+@property (nonatomic, assign) BOOL deferringUpdates;
+
+@end
+
+
+@implementation APCDisplacementTrackingCollector
+
+- (instancetype)initWithIdentifier:(NSString*)identifier deferredUpdatesTimeout:(NSTimeInterval) __unused anUpdateTimeout
+{
+    APCLogDebug(@"Initalizing location tracker");
+    
+    self = [super initWithIdentifier:identifier dateAnchorName:nil launchDateAnchor:nil];
+    
+    if (self != nil)
+    {
+        _deferringUpdates = NO;
+        
+        [self setupInitialLocationParameters];
+    }
+    
+    return self;
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)setupInitialLocationParameters
+{
+    _baseTrackingLocation       = [[CLLocation alloc] initWithLatitude:0.0 longitude:0.0];
+    _mostRecentUpdatedLocation  = [[CLLocation alloc] initWithLatitude:0.0 longitude:0.0];
+}
+
+- (void)start
+{
+    if ([CLLocationManager locationServicesEnabled] == YES)
+    {
+        if (!self.locationManager)
+        {
+            APCLogDebug(@"Start location tracking");
+            
+            self.locationManager            = [[CLLocationManager alloc] init];
+            self.locationManager.delegate   = self;
+            
+            if ([CLLocationManager significantLocationChangeMonitoringAvailable] == YES &&
+                [CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedAlways)
+            {
+                APCLogDebug(@"Significant Location Change Monitoring is Available");
+                
+                [self.locationManager startMonitoringSignificantLocationChanges];
+            }
+        }
+    }
+}
+
+- (void)stop
+{
+    if ([CLLocationManager locationServicesEnabled] == YES)
```

Comment:
@jwarmkessel The comparison is not safe on 32-bit platforms (iPhone 4s). The same comment applies to the next if statement.